Sources/CacheAPI-sqlite.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2017 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 3
+ */
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+/**
+ * SQLite Cache API class
+ * @package cacheAPI
+ */
+class sqlite_cache extends cache_api
+{
+  /**
+   * @var string The path to the current $cachedir directory.
+   */
+  private $cachedir		= null;
+  private $cacheDB		= null;
+  private $cacheTime	= null;
+
+	public function __construct()
+	{
+		global $modSettings;
+
+		parent::__construct();
+
+		// Set our default cachedir.
+		$this->setCachedir();
+
+		$database = $this->cachedir.'/'.'SQLite3Cache.db3';
+
+		$this->cacheDB = new SQLite3($database);
+		$this->cacheDB->busyTimeout(1000);
```

Comment:
Instead of connecting here in the __construct(), I suggest using the connect() method instead.  __construct() is called every time the API is called, while connect() is only called when we actually intend to use it.  __construct would also get called for when visiting the cache settings page as well.